Sources/Workspace/Workspace.swift

```diff
@@ -273,26 +322,26 @@ private struct WorkspaceDependencyResolverDelegate: DependencyResolverDelegate {
 }
 
 private class WorkspaceBinaryArtifactsManagerDelegate: Workspace.BinaryArtifactsManager.Delegate {
-    private unowned let workspaceDelegate: Workspace.Delegate
+    private weak var workspaceDelegate: Workspace.Delegate?
```

Comment:
not sure I understand this change, the delegate is not optional

Reply:
the delegate may be gone by the time the delegates methods are called, as they are on a queue. `weak` makes it safer than `unowned` since `unowned` means you control the lifetime which we dont.

Reply:
Ah I see, that makes sense.